### PR TITLE
Feature/us4239 create group functionality

### PR DESCRIPTION
**Pull request by Doug-Shannon**
Abililty to create a group and populate with current profile data - (not actually save a group.) Formly stuff was refined slightly as well.

### CODE REVIEW

File: crossroads.net/app/formlyBuilder/formlyBuilder.run.js [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+    angular.forEach(bindings, function (binding) {
+      ngModelAttrs[camelize(binding)] = { bound: binding };
+    });
+
+    console.log(ngModelAttrs);
+
+    formlyConfig.setType({
+      name: 'datepicker',
+      templateUrl: 'templates/datepicker.html',
+      //template: '<p class="input-group"><input  type="text"id="{{::id}}"name="{{::id}}"ng-model="model[options.key]"class="form-control"ng-click="datepicker.open($event)"uib-datepicker-popup="{{to.datepickerOptions.format}}"is-open="datepicker.opened"datepicker-options="to.datepickerOptions" /><span class="input-group-btn"><button type="button" class="btn btn-default" ng-click="datepicker.open($event)" ng-disabled="to.disabled"><i class="glyphicon glyphicon-calendar"></i></button></span></p>',
```
Comment (sseissiger) [inline crossroads.net/app/formlyBuilder/formlyBuilder.run.js]: please remove comment, unless we need it later :-)

File: crossroads.net/app/group_tool/create_group/createGroup.controller.js [hunk 2]
```diff
@@ -1,16 +1,18 @@
 export default class CreateGroupController {
     /*@ngInject*/
-    constructor(ParticipantService, $location, $log) {
+    constructor(ParticipantService, $location, $log, CreateGroupService) {
         this.log = $log;
         this.log.debug("CreateGroupController constructor");
         this.location = $location;
```
Comment (sseissiger) [inline crossroads.net/app/group_tool/create_group/createGroup.controller.js]: not sure if

File: crossroads.net/app/group_tool/services/createGroup.service.js [hunk 3]
```diff
@@ -0,0 +1,361 @@
...
+                    options: []
+                },
+                controller: /* @ngInject */ function ($scope, GroupService) {
+                    $scope.to.loading = GroupService.getSites().then(function (response) {
+                        $scope.to.options = response;
+                        return response;
+                    });
+                }
+            }, {
+                    key: 'profile.birthDate',
```
Comment (sseissiger) [inline crossroads.net/app/group_tool/services/createGroup.service.js]: Are we pre-populating birth date with profile data?

File: crossroads.net/app/group_tool/services/createGroup.service.js [hunk 4]
```diff
@@ -0,0 +1,361 @@
...
+                    labelProp: 'label',
+                    inline: false,
+                    options: [{
+                        label: 'Specific Day and Time',
+                    }, {
+                            label: 'Flexible Meeting Times/Not Sure Yet',
+                        }]
+                }
+            }, {
+                    key: 'group.meeting.day',
```
Comment (sseissiger) [inline crossroads.net/app/group_tool/services/createGroup.service.js]: As silly as it sounds meeting day should be coming from a lookup table

File: crossroads.net/app/group_tool/services/createGroup.service.js [hunk 5]
```diff
@@ -0,0 +1,361 @@
...
+                    }
+                }, {
+                    key: 'group.meeting.time',
+                    type: 'timepicker',
+                    templateOptions: {
+                        label: 'Time',
+                        minuteStep: 15
+                    }
+                }, {
+                    key: 'group.meeting.frequency',
```
Comment (sseissiger) [inline crossroads.net/app/group_tool/services/createGroup.service.js]: These options should also be coming from the database and filtered to just the two options for group_tool.

File: crossroads.net/app/group_tool/services/createGroup.service.js (hunk 3 above)
Comment (Doug-Shannon) [inline crossroads.net/app/group_tool/services/createGroup.service.js]: Yes, it is being handled in the constructor function on line 14 of this file.

**Pull request merged**